tools/pipeline-generator/Azure.Sdk.Tools.PipelineGenerator/PipelineGenerationContext.cs

```diff
@@ -161,6 +169,83 @@ public async Task<ServiceEndpointHttpClient> GetServiceEndpointClientAsync(Cance
             return cachedServiceEndpoint;
         }
 
+        public async Task<IEnumerable<ServiceEndpoint>> GetServiceConnectionsAsync(IEnumerable<string> serviceConnections, CancellationToken cancellationToken)
+        {
+            var serviceEndpointClient = await GetServiceEndpointClientAsync(cancellationToken);
+            var projectReference = await GetProjectReferenceAsync(cancellationToken);
+
+            var allServiceConnections = await serviceEndpointClient.GetServiceEndpointsAsync(projectReference.Id.ToString(), cancellationToken: cancellationToken);
+
+            this.logger.LogDebug("Returned a total of {Count} service endpoints", allServiceConnections.Count);
+            
+            List<ServiceEndpoint> endpoints = new List<ServiceEndpoint>();
+            foreach (var serviceConnection in allServiceConnections)
+            {
+                if (serviceConnections.Contains(serviceConnection.Name))
+                {
+                    endpoints.Add(serviceConnection);
+                }
+            }
+            return endpoints;
+        }
+
+        private HttpClient cachedRawHttpClient = null;
+
+        private async Task<HttpClient> GetRawHttpClient(CancellationToken cancellationToken)
+        {
+            if (this.cachedRawHttpClient == null)
+            {
+                var credential = GetAzureCredentials();
+                var tokenRequestContext = new TokenRequestContext(new[] { "499b84ac-1321-427f-aa17-267ca6975798/.default" });
```

Comment:
Can we make this a named const so it's clear what it is?

Reply:
I'll add a comment.